plugins/modules/aci_oob_contract.py

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        contract=dict(type="str", aliases=["contract_name", "name"]),  # Not required for querying all objects
+        description=dict(type="str", aliases=["descr"]),
+        scope=dict(type="str", choices=["application-profile", "context", "global", "tenant"]),
+        priority=dict(type="str", choices=["level1", "level2", "level3", "unspecified"]),  # No default provided on purpose
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["contract"]],
+            ["state", "present", ["contract"]],
+        ],
+    )
+
+    contract = module.params.get("contract")
+    description = module.params.get("description")
+    scope = module.params.get("scope")
+    priority = module.params.get("priority")
+    state = module.params.get("state")
+    tenant = "mgmt"  # hard coding tenant to 'mgmt'
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="vzOOBBrCP",
+            aci_rn="oobbrc-{0}".format(contract),
+            module_object=contract,
+            target_filter={"name": contract},
+        ),
+    )
```

Comment:
Can we define the subclass combine to be able to leverage class query for query all?

```suggestion
    name_alias = module.params.get("name_alias")

    aci = ACIModule(module)
    aci.construct_url(
        root_class=dict(
            aci_class="vzOOBBrCP",
            aci_rn="tn-mgmt/oobbrc-{0}".format(contract),
            module_object=contract,
            target_filter={"name": contract},
        ),
    )
```

Reply:
Would have to test this once for querying operations. Will analyse over the weekend.

Reply:
done, all tests pass as well.
Marking resolved